Dockerfile

```
@@ -6,23 +6,61 @@ ARG DEBIAN_FRONTEND=noninteractive
 
 ENV PYTHONUNBUFFERED 1
 
+# https://python-poetry.org/docs/configuration/#using-environment-variables
+ENV POETRY_VERSION=1.1.0 \
+    POETRY_HOME="/opt/poetry" \
+    POETRY_VIRTUALENVS_IN_PROJECT=true \
+    POETRY_NO_INTERACTION=1 \
+    PYSETUP_PATH="/opt/pysetup" \
+    VENV_PATH="/opt/pysetup/.venv"
+
+# prepend poetry and venv to path
+ENV PATH="$POETRY_HOME/bin:$VENV_PATH/bin:$PATH"
+
+FROM python-base as builder-base
+
 RUN apt-get update && apt-get install -y \
     build-essential python3-dev \
     libldap2-dev libsasl2-dev ldap-utils tox \
     lcov valgrind vim \
     && pip3 install uwsgi
 
+RUN pip3 install --upgrade pip setuptools wheel
+
+# Install Poetry - respects $POETRY_VERSION & $POETRY_HOME
+RUN curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py | python3
+
+# copy project requirement files here to ensure they will be cached.
+WORKDIR $PYSETUP_PATH
+COPY pyproject.toml poetry.lock .
```

```suggestion
COPY pyproject.toml poetry.lock ./
```

Why?
- As seen on https://github.com/TACC/Core-Portal/blob/main/server/conf/docker/Dockerfile#L45.
- Should resolve https://jenkins01.tacc.utexas.edu/job/Core_CMS_Build/463/console failure.
    ```
    Step 12/26 : COPY pyproject.toml poetry.lock .
    When using COPY with more than one source file, the destination must be a directory and end with a /
    ```